databricks/koalas/groupby.py

```diff
@@ -0,0 +1,312 @@
+    _MissingPandasLikeSeriesGroupBy
+from databricks.koalas.series import Series, _col
+
+
+ColumnLike = Union[str, Series]
+
+
+class GroupBy(object):
+
+    def __new__(cls, obj: _Frame, *args, **kwargs):
+        if isinstance(obj, DataFrame):
+            return super(GroupBy, cls).__new__(DataFrameGroupBy)
+        elif isinstance(obj, Series):
+            return super(GroupBy, cls).__new__(SeriesGroupBy)
+        else:
+            raise TypeError('Constructor expects DataFrame or Series; however, '
+                            'got [%s]' % (obj,))
+
+    # TODO: Series support is not implemented yet.
+    # TODO: not all arguments are implemented comparing to Pandas' for now.
+    def aggregate(self, func_or_funcs, *args, **kwargs):
+        """Aggregate using one or more operations over the specified axis.
+
+        Parameters
+        ----------
+        func : dict
+             a dict mapping from column name (string) to aggregate functions (string).
+
+        Returns
+        -------
+        Series or DataFrame
+
+            The return can be:
+
+            * Series : when DataFrame.agg is called with a single function
+            * DataFrame : when DataFrame.agg is called with several functions
+
+            Return Series or DataFrame.
+
+        Notes
+        -----
+        `agg` is an alias for `aggregate`. Use the alias.
+
+        Examples
+        --------
+
+        >>> df = ks.DataFrame({'A': [1, 1, 2, 2],
+        ...                    'B': [1, 2, 3, 4],
+        ...                    'C': [0.362, 0.227, 1.267, -0.562]})
+
+        >>> df
+           A  B      C
+        0  1  1  0.362
+        1  1  2  0.227
+        2  2  3  1.267
+        3  2  4 -0.562
+
+        Different aggregations per column
+
+        >>> df.groupby('A').agg({'B': 'min', 'C': 'sum'})
+           B      C
+        0  1  0.589
+        1  3  0.705
+
+        """
+        if not isinstance(func_or_funcs, dict) or \
+            not all(isinstance(key, string_types) and isinstance(value, string_types)
+                    for key, value in func_or_funcs.items()):
+            raise ValueError("aggs must be a dict mapping from column name (string) to aggregate "
+                             "functions (string).")
+
+        sdf = self._kdf._sdf
+        groupkeys = self._groupkeys
+        groupkey_cols = [s._scol.alias('__index_level_{}__'.format(i))
+                         for i, s in enumerate(groupkeys)]
+        gdf = sdf.groupby(*groupkey_cols).agg(func_or_funcs)
+        reordered = ['%s(%s)' % (value, key) for key, value in iter(func_or_funcs.items())]
+        kdf = DataFrame(gdf.select(reordered))
+        kdf.columns = [key for key in iter(func_or_funcs.keys())]
+
+        return kdf
+
+    agg = aggregate
+
+    def count(self):
+        """
+        Compute count of group, excluding missing values.
+        """
+        return self._reduce_for_stat_function(F.count, only_numeric=False)
```

Comment:
do we need to handle nan values differently?

Reply:
I think that's handled in `_reduce_for_stat_function`. Yup, it's different from what we do in `frame.py`. Let me at least add a doctest here for now.